pkg/build/builder/cmd/builder.go

```diff
@@ -41,22 +41,28 @@ func run(builderFactory factoryFunc) {
 		authcfg     docker.AuthConfiguration
 		authPresent bool
 	)
+	output := true
 	if len(build.Parameters.Output.DockerImageReference) == 0 {
 		if build.Parameters.Output.To != nil {
```

Comment:
With this condition you'll still end build when `To` is not defined and if I understood @smarterclayton correctly he wanted to have no output defined at all.